### ISSUE TITLE
documentation of VariableDeclarationUsageDistance has stange note of default distance

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#9219]
https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance_Notes

`As distance by default is 1 the ....`

`As the distance is more than the default one, ....`

default `allowedDistance` is `3` https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance_Properties

it is not clear what doc mean, or it might be outdated.

**Comment by romani** [comment]
@bossever , please help to resolve this issue, it was detected during review of https://github.com/checkstyle/checkstyle/pull/9215

**Comment by bossever** [comment]
@romani, after checking the examples in **Unsupported Cases** I think the check is working exactly as intended, this section seems to be outdated. I think it can be removed so that it does not confuse the user about the working of the check.

**Comment by bossever** [comment]
### Case 1: ###
`$ cat Test.java`
```java
public class Test {

    public void foo() {
        int c;            // OK, distance = 2
        int a = 3;      // OK, distance = 1
        int b = 2;      // OK, distance = 1
        {
          a = a + b;
          c = b;
        }
    }
} 
```
`$ cat check.xml`
```xml
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">		
  <module name="TreeWalker">
    <module name="VariableDeclarationUsageDistance">
      <property name="validateBetweenScopes" value="true"/>
    </module>
  </module>
</module>
```

`$ java com.puppycrawl.tools.checkstyle.Main -c`
```
Starting audit...
Audit done.
```

As expected, the check does not raise violation because maximum allowed distance = 3.

### Case 2: ###
`$ cat Test.java`
```java
public class Test {
    boolean res;

    public void foo() {
        int sum = 0;            // OK, distance = 3
        
        for (int i = 0; i < 20; i++) {
            a++;
            b--;
            sum++;
            
            if (sum > 10) {
              res = true;
            }
        }
    }
}
```
`$ cat check.xml`
```xml
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">		
  <module name="TreeWalker">
    <module name="VariableDeclarationUsageDistance">
      <property name="validateBetweenScopes" value="true"/>
    </module>
  </module>
</module>
```

`$ java com.puppycrawl.tools.checkstyle.Main -c`
```
Starting audit...
Audit done.
```

As expected, the check does not raise violation because max allowed distance = 3, and in the example also, distance between usage and declaration for `sum` = 3.

**Comment by bossever** [comment]
@romani , I have created the PR but it does not come up "Linked pull requests" on the side bar, although I have mentioned this issue with the link in the PR.
#9224

**Comment by romani** [comment]
fix is merged
